pkg/controller/pod_volume_restore_controller.go

```diff
@@ -262,38 +249,26 @@ func (c *PodVolumeRestoreReconciler) processRestore(ctx context.Context, req *ve
 		return errors.Wrap(err, "error getting backup storage location")
 	}
 
-	// if there's a caCert on the ObjectStorage, write it to disk so that it can be passed to restic
-	var caCertFile string
-	if backupLocation.Spec.ObjectStorage != nil && backupLocation.Spec.ObjectStorage.CACert != nil {
-		caCertFile, err = restic.TempCACertFile(backupLocation.Spec.ObjectStorage.CACert, req.Spec.BackupStorageLocation, c.fileSystem)
-		if err != nil {
-			log.WithError(err).Error("Error creating temp cacert file")
-		}
-		// ignore error since there's nothing we can do and it's a temp file.
-		defer os.Remove(caCertFile)
+	backupRepo, err := crd.GetBackupRepositoryByFilter(ctx, c.Client, req.Namespace, req.Spec.Pod.Namespace, req.Spec.BackupStorageLocation)
```

Comment:
The `PodVolumeBackup` and `PodVolumeRestore` should contain enough information about where the data is put to or get from.
Currently, there is a `RepoIdentifier` in PVB/PVR, if it isn't enough, we should add more info rather than guessing the repo name

Reply:
`RepoIdentifier` is not enough since we have two path, the Restic path and the unified path. If we have two `BackupRepository` for the two paths representatively but they are from the same storage config, `RepoIdentifier` will be the same.
Moreover, at present, `RepoIdentifier` is used by Restic path only.

Maybe we can follow the way in `func (r *RepositoryEnsurer) EnsureRepo` to retrieve the `BackupRepository` object?

Reply:
In `func (r *RepositoryEnsurer) EnsureRepo`, it will triger to create related `BackupRepository` and watch the status then retrieve the `BackupRepository `

Here, we don't do anything to update the related `BackupRepository`. Maybe it's not proper.
And also it's a little complicated when you just want to get related `BackupRepository` by watching resources.

Here I just want to get `BackupRepository`, The directest way is using `kubeclient.Get(ctx, NamespacedName)`, we know the namespace, and if we can store Repository Name on some CRD will be good

Reply:
I mean follow the way in the function to get the `BackupRepository`, that is, use the label selector:
```
	selector := labels.SelectorFromSet(repoLabels(volumeNamespace, backupLocation))
	repos, err := r.repoLister.BackupRepositories(namespace).List(selector)
```

Reply:
I agree with @qiuming-best, adding the backup repo name to the PVB/PVR may be more reasonable? The name is unique and can identify the repository

Reply:
Now, I've changed into `client.List(ctx, backupRepoList, &client.ListOptions{
		Namespace:     pvbNamespace,
		LabelSelector: selector,
	})`
the label selector which include `volumeNamespace`, `BackupStorageLocation`, `repositoryType`

Reply:
@ywk253100 Let PVB/PVR directly refer to the BackupRepository name doesn't solve all the problems -- for old PVB/PVR, they don't have this reference, on the other hand, multiple PVB/PVR may refer to one repository, so below situation may happen:

- Two old PVBs refer to one ResticRepository
- When we start PVB1, since its repo reference is empty, we fail back to create a BackupRepository for Restic
- When we start PVB2, its repo reference is also empty, but this time, we should find a way to search the existing BackupRepository instead of creating a new one

As you can see, we still need a method of "Search".

Moreover, as the previous discussion, our current approach to handle BackupRepository is not enough for Unified Repo when there are multiple repo providers, besides kopia, so we are using some hardcode things (for details, refer to [this comment](https://github.com/vmware-tanzu/velero/pull/5259#discussion_r959516173)). This means, in future, we will face refactor of these areas, therefore, the good enough and simpler solution is preferred.